blog/2023/12/julia-1.10-highlights.md

```diff
@@ -0,0 +1,105 @@
++++
+mintoclevel = 2
+maxtoclevel = 3
+title = "Julia 1.10 Highlights"
+authors = "The Julia contributors"
+published = "22 Dec 2023"
+rss_pubdate = Date(2023, 12, 22)
+rss = """Highlights of the Julia 1.10 release."""
++++
+
+
+After X betas and X release candidates, Julia version 1.10 has finally(!) been released. We would like to thank all the contributors to this release and all the testers that helped with finding regressions and issues in the pre-releases. Without you, this release would not have been possible.
+
+The full list of changes can be found in the [NEWS file](https://github.com/JuliaLang/julia/blob/release-1.9/NEWS.md), but here we'll give a more in-depth overview of some of the release highlights.
+
+\toc
+
+# New parser written in Julia
+
+*Claire Foster*
+
+In Julia 1.10 we have replaced Julia's default parser (written in Scheme) with one written in Julia...
+
+# Package load time improvements
+
+*Jameson Nash et al.*
+
+In 1.9 the TTFX was heavily improved by allowing package authors to enable saving of native code during precompilation.
+
+In 1.10, as a follow up, signficant work was put into the performance of the loading of packages
+
+A lot of this work was driven by profiling and improving the load time of [OmniPackage.jl](https://github.com/JuliaComputing/OmniPackage.jl) which is an artifical "mega package" which only purpose is to depend on and load a lot of dependencies.
+
+- Improvement to the type system... (Jameson)
+- Reudction of invalidations causing recompilation
+- Moving packages away from Requires.jl to package extensions
+- `mul!` dispatch improvements (Daniel K)
+
+
+# Stacktrace rendering improvements
+
+*Jeff Bezanson, Tim Holy*
+
+- `Type{...}` in stacktraces
+- Successive frames at identical location collapsed
+- https://github.com/JuliaLang/julia/pull/49102
+
+
+# Parallel GC
+
+*Diogo Netto*
+
+Up to 1.9, Julia's garbage collector was stop-the-world (meaning that all compute threads had to be stopped while a collection cycle was running) and serial (meaning a single thread performed a collection while the compute threads were halted). As one may imagine, garbage collection can quickly become a bottleneck in multithreaded code, particularly in code where multiple threads allocate.
```

Comment:
"Up to 1.9, ..." feels a bit misleading here since (most of) the GC is still stop-the-world in 1.10.

Reply:
Agreed. Plan to update this very soon.